gql/resolver_sealingpipeline.go

```diff
@@ -75,36 +75,55 @@ func (r *resolver) SealingPipeline(ctx context.Context) (*sealingPipelineState,
 		}
 	}
 
-	// TODO: put states in order
-	var order int32
 	sectorStates := []*sectorState{}
-	for k, v := range summary {
-		if v == 0 {
+	for order, state := range allSectorStates {
+		count, ok := summary[api.SectorState(state)]
+		if !ok {
+			continue
+		}
+		if count == 0 {
 			continue
 		}
-		order++
 
-		if _, ok := normalSectors[string(k)]; ok {
+		if _, ok := normalSectors[state]; ok {
 			sectorStates = append(sectorStates, &sectorState{
-				Key:   string(k),
-				Value: int32(v),
+				Key:   state,
+				Value: int32(count),
 				Type:  "Regular",
-				Order: order,
+				Order: int32(order),
 			})
 			continue
 		}
 
-		if _, ok := snapdealsSectors[string(k)]; ok {
+		if _, ok := normalErredSectors[state]; ok {
 			sectorStates = append(sectorStates, &sectorState{
```

Comment:
Not sure if we want to bundle all sectors into one `sectorStates` struct, and split on the frontend...

or immediately send 4 separate structs to the frontend.